lib/Parser/BGParseManager.cpp

```diff
@@ -0,0 +1,467 @@
+bool BGParseManager::WasAddedToJobProcessor(JsUtil::Job *const job) const
+{
+    Assert(!"BGParseManager::WasAddedToJobProcessor");
+    return true;
+}
+
+
+// Note: runs on any thread
+BGParseWorkItem::BGParseWorkItem(
+    BGParseManager* manager,
+    const byte* pszScript,
+    size_t cbScript,
+    char16 *fullPath
+    )
+    : JsUtil::Job(manager),
+    script(pszScript),
+    cb(cbScript),
+    path(nullptr),
+    parseHR(S_OK),
+    parseSourceLength(0),
+    bufferReturn(nullptr),
+    bufferReturnBytes(0),
+    complete(nullptr)
+{
+    this->cookie = BGParseManager::GetNextCookie();
+
+    Assert(fullPath != nullptr);
+    this->path = SysAllocString(fullPath);
+}
+
+BGParseWorkItem::~BGParseWorkItem()
+{
+    SysFreeString(this->path);
+    if (this->complete != nullptr)
+    {
+        HeapDelete(this->complete);
+    }
+}
+
+void BGParseWorkItem::TransferCSE(CompileScriptException* pse)
+{
+    this->cse.CopyInto(pse);
+}
+
+// This function parses the input data cached in BGParseWorkItem and caches the
+// seralized bytecode
+// Note: runs on BackgroundJobProcessor thread
+// Note: All exceptions are caught by BackgroundJobProcessor
+void BGParseWorkItem::ParseUTF8Core(Js::ScriptContext* scriptContext)
+{
+    if (PHASE_TRACE1(Js::BgParsePhase))
+    {
+        Js::Tick now = Js::Tick::Now();
+        Output::Print(
+            _u("[BgParse: Parse -- cookie: %04d on thread 0x%X at %.2f ms]\n"),
+            GetCookie(),
+            ::GetCurrentThreadId(),
+            now.ToMilliseconds()
+        );
+    }
+
+    Js::AutoDynamicCodeReference dynamicFunctionReference(scriptContext);
+    SourceContextInfo* sourceContextInfo = scriptContext->GetSourceContextInfo(this->cookie, nullptr);
+    if (sourceContextInfo == nullptr)
+    {
+        sourceContextInfo = scriptContext->CreateSourceContextInfo(this->cookie, this->path, wcslen(this->path), nullptr);
+    }
+
+    SRCINFO si = {
+        /* sourceContextInfo   */ sourceContextInfo,
+        /* dlnHost             */ 0,
+        /* ulColumnHost        */ 0,
+        /* lnMinHost           */ 0,
+        /* ichMinHost          */ 0,
+        /* ichLimHost          */ (ULONG)0,
+        /* ulCharOffset        */ 0,
+        /* mod                 */ 0,
+        /* grfsi               */ 0
+    };
+
+    // Currently always called from a try-catch
+    ENTER_PINNED_SCOPE(Js::Utf8SourceInfo, sourceInfo);
+    sourceInfo = Js::Utf8SourceInfo::NewWithNoCopy(scriptContext, (LPUTF8)this->script, (int32)this->cb, static_cast<int32>(this->cb), &si, false);
+    LEAVE_PINNED_SCOPE();
+
+    // what's a better name for "fIsOriginalUtf8Source?" what if i want to say "isutf8source" but not "isoriginal"?
+    charcount_t cchLength = 0;
+    uint sourceIndex = 0;
+    Js::ParseableFunctionInfo * func = nullptr;
+    this->parseHR = scriptContext->CompileUTF8Core(sourceInfo, &si, true, this->script, this->cb, 0x1000, &this->cse, cchLength, this->parseSourceLength, sourceIndex, &func);
```

Comment:
>0x1000 [](start = 98, length = 6)

Could we use the name of this value instead of the value directly? Looks like this is `fscrGlobalCode` unless I'm misinterpreting the usage.

Reply:
yes--remnants of prototype